packages/datetime/test/datePickerTests.tsx

```diff
@@ -26,6 +26,13 @@ describe("<DatePicker>", () => {
         assert.isUndefined(root.state("selectedDay"));
     });
 
+    it("week starts with firstDayOfWeek value", () => {
+        const selectedFirstDay = 3;
+        const wrapper = mount(<DatePicker dayPickerBaseProps={{ firstDayOfWeek: selectedFirstDay }} />);
+        const firstWeekday = wrapper.find("Weekday").first();
+        assert.equal(firstWeekday.prop("weekday"), selectedFirstDay);
+    });
```

Comment:
Will need tests for each of the weird cases I mentioned in my big comment above.